overlord/ifacestate/helpers.go

```diff
@@ -102,6 +108,20 @@ func (m *InterfaceManager) addSnaps() error {
 	return nil
 }
 
+func (m *InterfaceManager) profilesNeedRegeneration() bool {
+	raw, err := ioutil.ReadFile(dirs.SnapSystemKeyFile)
+	if os.IsNotExist(err) {
+		return true
+	}
+	if err != nil {
+		logger.Noticef("cannot read system-key file: %s", err)
+		return true
+	}
+
+	onDiskSystemKey := string(raw)
+	return onDiskSystemKey != interfaces.SystemKey()
```

Comment:
Can we de-serialize it and compare the build-id == "unknown" as distinct from everything?